src/constraints/constraint_units_available.jl

```diff
@@ -53,10 +53,15 @@ function _build_constraint_units_available(m, u, s, t)
             init=0,
         )
         <=
-        number_of_units(m; unit=u, stochastic_scenario=s, analysis_time=t0, t=t)
+        number_of_units(m; unit=u, stochastic_scenario=s, analysis_time=t0, t=t,
+            _default=_default_number_of_units(u))
     )
 end
 
+# Change the default number of units so that it is zero when candidate units are present
+# and otherwise 1.
+_default_number_of_units(u) = is_candidate(unit=u) ? 0 : 1
```

Comment:
Should the `1` be `number_of_units(...)` instead?

Reply:
Also, unit tests?

Reply:
I now changed the code to reflect this and updated the unit test.